### PR TITLE
John conroy/provenance testing

### LINKED ISSUE #1609: Don't display donor registration node
The first node in the graph should be `Donor`: Neither the fake root node, nor the donor registration activity are useful.
Screen Shot 2021-03-12 at 9 16 49 AM

### PR DESCRIPTION
**Pull request by john-conroy**
Add testing for provenance graph.

We also have three gitignores throughout the repo. I updated them to ignore the screenshots produced by cypress. Unsure if they're all needed.

From #1652: Fix #1609.

### CODE REVIEW

File: context/app/static/js/components/Detail/provenance/ShowDerivedEntitiesButton/ShowDerivedEntitiesButton.spec.js [hunk 1]
```diff
@@ -0,0 +1,8 @@
+// import { render, screen } from 'test-utils/functions';
+import { getUniqueNewSteps } from './ShowDerivedEntitiesButton';
+
+test('should return only unique new steps', () => {
+  const existingSteps = [{ name: 'a' }, { name: 'b' }, { name: 'c' }];
+  const newSteps = [{ name: 'a' }, { name: 'c' }, { name: 'd' }];
+  expect(getUniqueNewSteps(existingSteps, newSteps)).toEqual([{ name: 'd' }]);
+});
```
Comment (ilan-gold) [inline context/app/static/js/components/Detail/provenance/ShowDerivedEntitiesButton/ShowDerivedEntitiesButton.spec.js:8]: Why not test the component?

**Review by ilan-gold — APPROVED** [review]
Can you merge `master` and make sure everything still works?

File: context/app/static/js/components/Detail/provenance/ProvGraph/ProvGraph.spec.js [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+    'Create Sample Activity - HBM665.NTZB.997',
+    sampleEntityText,
+  ];
+
+  render(<ProvGraph provData={sampleProv} />);
+
+  nodesText.forEach((text) => expect(screen.getByText(text)).toBeInTheDocument());
+});
+
+test('should selected node information in detail pane and show immediate descendants when show derived entities button is clicked', async () => {
```
Comment (ilan-gold) [inline context/app/static/js/components/Detail/provenance/ProvGraph/ProvGraph.spec.js:66]:
```suggestion
test('should select node information in detail pane and show immediate descendants when showing derived entities button is clicked', async () => {
```
selected -> select and show -> showing?

Comment (john-conroy) [reply context/app/static/js/components/Detail/provenance/ProvGraph/ProvGraph.spec.js:66]: 'should display selected node information in detail pain and show immediate descendants when "Show Derived Entities" button is clicked', perhaps?

File: context/app/static/js/components/Detail/provenance/ShowDerivedEntitiesButton/ShowDerivedEntitiesButton.spec.js (hunk 1 above)
Comment (john-conroy) [reply context/app/static/js/components/Detail/provenance/ShowDerivedEntitiesButton/ShowDerivedEntitiesButton.spec.js:8]: I could test that it makes the correct calls to the APIs?

File: context/app/static/js/components/Detail/provenance/ProvGraph/ProvGraph.spec.js (hunk 2 above)
Comment (ilan-gold) [reply context/app/static/js/components/Detail/provenance/ProvGraph/ProvGraph.spec.js:66]: sounds good!

File: context/app/static/js/components/Detail/provenance/ShowDerivedEntitiesButton/ShowDerivedEntitiesButton.spec.js (hunk 1 above)
Comment (ilan-gold) [reply context/app/static/js/components/Detail/provenance/ShowDerivedEntitiesButton/ShowDerivedEntitiesButton.spec.js:8]: Hmm, i guess there isn't really much going to test actually.  Feel free to ignore.

**Pull request merged**